.github/workflows/cd.yml

```diff
@@ -0,0 +1,59 @@
+name: CD
+
+on:
+  push:
+    branches: [ "master" ]
+    tags: [ "*" ]
+
+jobs:
+  release:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v2
+
+      - name: Setup dependencies
+        run: npm install
+
+      - name: Build xsnippet-web
+        run: npm run build
+
+      - name: Create xsnippet-web.tar.gz
+        run: tar cvzf xsnippet-web.tar.gz -C dist/ .
+
+      - name: Setup release metadata
+        run: |
+          PRERELEASE=true
+          VERSION=nightly
+
+          if [[ $GITHUB_REF =~ ^refs/tags/.* ]]; then
+            if [[ $GITHUB_REF =~ ^refs/tags/\d+.\d+.\d+$ ]]; then
+              PRERELEASE=false
+            fi
+            VERSION="${GITHUB_REF#refs/tags/}"
+          fi
+
+          echo ::set-output name=version::${VERSION}
+          echo ::set-output name=prerelease::${PRERELEASE}
+        id: metadata
+
+      # Since 'softprops/action-gh-release@v1' does not support release
+      # re-creation, we have to remove 'nightly' release first.
+      - name: Remove nightly release
```

Comment:
I feel like I'm going to write my own GitHub action for the whole "create release" thing :D